src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -100,7 +100,7 @@ public override void Write(byte[] buffer, int offset, int count)
                 throw new ArgumentOutOfRangeException("count");
             }
 
-            if (offset + count > buffer.Length)
+            if ((long)offset + (long)count > (long)buffer.Length)
```

Comment:
Thanks, similar code in WinHttpResponseStream.cs was already using these casts so that's the reason I went and used longs..